gremlin-go/driver/connection_test.go

```diff
@@ -203,9 +196,35 @@ func TestConnection(t *testing.T) {
 			assert.NotNil(t, resultSet)
 			result := resultSet.one()
 			assert.NotNil(t, result)
-			assert.Equal(t, "[0]", result.GetString())
+			assert.Equal(t, "0", result.GetString())
 			err = client.Close()
 			assert.Nil(t, err)
 		}
 	})
+
+	t.Run("Test DriverRemoteConnection Next and HasNext", func(t *testing.T) {
+		if runIntegration {
+			// Add data
+			remote, err := NewDriverRemoteConnection(testHost, testPort)
+			assert.Nil(t, err)
+			assert.NotNil(t, remote)
+			g := Traversal_().WithRemote(remote)
+			dropGraph(t, g)
+			addTestData(t, g)
+
+			// Run traversal and test Next/HasNext calls
+			traversal := g.V().HasLabel(personLabel).Properties(nameKey)
+			var names []string
+			for i := 0; i < len(getTestNames()); i++ {
+				hasN, _ := traversal.HasNext()
+				assert.True(t, hasN)
+				res, _ := traversal.Next()
+				vp, _ := res.GetVertexProperty()
```

Comment:
You should always read errors out and check them in tests.
```suggestion
				hasN, err := traversal.HasNext()
				assert.Nil(t, err)
				assert.True(t, hasN)
				res, err := traversal.Next()
				assert.Nil(t, err)
				assert.NotNil(t, res)
				vp, err := res.GetVertexProperty()
				assert.Nil(t, err)
```